flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/PipelinedRegionSchedulingStrategy.java

```diff
@@ -226,44 +228,71 @@ private void maybeScheduleRegions(final Set<SchedulingPipelinedRegion> regions)
                 SchedulingStrategyUtils.sortPipelinedRegionsInTopologicalOrder(
                         schedulingTopology, regions);
```

Comment:
I'm not entirely sure whether this sorting can be excluded from the loop. It seems there's no guarantee that the downstream regions being found are in strict topological order. I don't see immediate damage of this, but we'd better double check.

Reply:
How about instead to do the sorting for `regionsToSchedule` before `scheduleRegion`?
This helps to guarantee that we schedule regions in topological order, in one scheduling batch.

Reply:
Change to sort `regionToSchedule` before `scheduleRegion`,  Jm log about task scheduling subject to topological order now.